modules/SofaGeneralMeshCollision/src/SofaGeneralMeshCollision/DirectSAP.h

```diff
@@ -49,25 +56,22 @@ class EndPoint;
   */
 class SOFA_SOFAGENERALMESHCOLLISION_API DSAPBox{
 public:
-    DSAPBox(Cube c,EndPoint * mi = nullptr,EndPoint * ma = nullptr) : cube(c),min(mi),max(ma){}
+    explicit DSAPBox(const Cube& c,EndPoint * mi = nullptr,EndPoint * ma = nullptr) : cube(c),min(mi),max(ma){}
```

Comment:
Yes, the compiler will allow to do that: 
```
Cube cube;
DSAPBox box = cube;
```
or
```
void f(DSAPBox box) {...}
Cube cube;
f(cube);
```
Adding the explicit keyword prevents the compiler to do this type of conversions. It is very useful to avoid some types of bugs difficult to find because implicit conversions are hard to detect.